hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -245,19 +224,18 @@ private TokenRelationship buildTokenRelationship(
 	private void persistNonDestroyed(
 			TokenRelationship modelRel,
 			EntityNumPair key,
-			MerkleMap<EntityNumPair, MerkleTokenRelStatus> currentTokenRels
+			BackingStore<Pair<AccountID, TokenID>, MerkleTokenRelStatus> currentTokenRels
 	) {
 		final var isNewRel = modelRel.isNotYetPersisted();
 		final var mutableTokenRel = isNewRel
 				? new MerkleTokenRelStatus()
-				: currentTokenRels.getForModify(key);
+				: currentTokenRels.getRef(key.asAccountTokenRel());
 		mutableTokenRel.setBalance(modelRel.getBalance());
 		mutableTokenRel.setFrozen(modelRel.isFrozen());
 		mutableTokenRel.setKycGranted(modelRel.isKycGranted());
 		mutableTokenRel.setAutomaticAssociation(modelRel.isAutomaticAssociation());
 		if (isNewRel) {
-			currentTokenRels.put(key, mutableTokenRel);
-			alertTokenBackingStoreOfNew(modelRel);
+			currentTokenRels.put(key.asAccountTokenRel(), mutableTokenRel);
```

Comment:
This `put()` needs to happen no matter if `isNewRel` is `true` or `false`. (Consider the case when the `BackingStore` is actually a `TransactionaLedger`.